ui/guest/src/classes/ContentController.ts

```diff
@@ -484,6 +484,27 @@ export function sortItem(
   // Insert in desired position
   result.splice(targetIndexParsed, 0, collection[currentIndexParsed]);
 
+  function updateModel(fieldId: string, model: ModelHierarchyDescriptor, currentTarget: number, targetIndex: number) {
+    const position = model.parentContainerFieldPath.split('.').indexOf(fieldId);
+    const index = model.parentContainerFieldIndex as string;
+    const splitIndex = index.split('.');
+    const numericIndex = Number(splitIndex[position]);
+
+    if (numericIndex === currentTarget) {
+      splitIndex[position] = targetIndex.toString();
+      model.parentContainerFieldIndex = splitIndex.join('.');
+    } else if (numericIndex === targetIndex) {
+      splitIndex[position] = currentTarget.toString();
+      model.parentContainerFieldIndex = splitIndex.join('.');
+    } else if (targetIndex < currentTarget) {
+      splitIndex[position] = (numericIndex + 1).toString();
+      model.parentContainerFieldIndex = splitIndex.join('.');
+    } else if (targetIndex > currentTarget) {
+      splitIndex[position] = (numericIndex + -1).toString();
+      model.parentContainerFieldIndex = splitIndex.join('.');
+    }
```

Comment:
```suggestion
    if (numericIndex === currentTarget) {
      splitIndex[position] = targetIndex.toString();
    } else if (numericIndex === targetIndex) {
      splitIndex[position] = currentTarget.toString();
    } else if (targetIndex < currentTarget) {
      splitIndex[position] = (numericIndex + 1).toString();
    } else if (targetIndex > currentTarget) {
      splitIndex[position] = (numericIndex - 1).toString();
    }

    model.parentContainerFieldIndex = splitIndex.join('.');
```

Reply:
Isn't this logic unnecessary? Whether current vs target index is greater or lesser, we're not modifying the item's index but the repeat container's index so, shouldn't this just set the same index for all?
```suggestion
    splitIndex[position] = String(targetIndex);
    model.parentContainerFieldIndex = splitIndex.join('.');
```

Reply:
let's discuss this

Reply:
Using the old approach, this needs to be reviewed